Refactor drawsvg.js

yxiong opened the pull request:
I made an attempt to refactor the `drawsvg.js` file such that the logic for different drawing style (`sunburst` vs `icicle`) are grouped together. To the best of my knowledge, the refactor should not affect any visual output in the browser, but feel free to let me know if it actually does.

BTW, this PR does not contain the change in #56.

Hope this is useful!

jiffyclub commented:
Many thanks for this! If you've followed the discussion in #55 I've been thinking about how to refactor the JS to make the visualizations more easily embeddable, so this is helpful.

The only thing I think I'd change here would be to take the content of the `get_render_params` function and split it into two functions `get_sunburst_render_params` and `get_icicle_render_params`, then have `get_render_params` call those. Then `get_render_params("sunburst")["radius"]` could be `get_sunburst_render_params()["radius"]`. I can do this later if you don't have time.

Really I should probably change that "radius" term to something else and put it in its own function since it comes up a bunch, but that can wait.

yxiong commented:
Yep, that makes sense. I split the `get_render_params` function as you suggested, and changed `get_render_params("sunburst")["radius"]` into `get_sunburst_render_params()["radius"]`. I also agree it would be better to change the `radius` term to something more meaningful, probably in the next round of refactor.

jiffyclub commented:
:+1: Thanks again!
